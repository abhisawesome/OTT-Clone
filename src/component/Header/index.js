
import { memo } from 'react';
import utils from '../../utils';
import { useDispatch } from 'react-redux';

const Header = () => {
    utils.log('Header rendered !');
    const dispatch = useDispatch();
    const searchFilim = (value) => {
        dispatch({
            type: 'search_movie',
            payload: {
                movieName: value
            }
        })
    }
    return (
        //bg-[url('/images/nav_bar.png')]
        <div className="flex items-center justify-between p-4 ">
            <div>
                <img
                    alt='logo'
                    className='h-5 w-5 sm:pl-30'
                    src="images/Back.png"
                />
            </div>

            <div className='flex item-center group' >
                <img
                    alt='logo'
                    className='h-5 w-5 pl-100 block group-hover:hidden'
                    src="images/search.png"
                />

                <input
                    className="text-black hidden group-hover:block"
                    type="text" placeholder="Search movie.."
                    onChange={(event) => {
                        searchFilim(event.target.value);
                    }}
                />
            </div>

        </div>
    )
}

export default memo(Header);