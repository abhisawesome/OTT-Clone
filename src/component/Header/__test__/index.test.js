import * as React from 'react'
import renderer from 'react-test-renderer';

import Header from '../index';
import { useSelector, useDispatch } from 'react-redux'

jest.mock('react-redux');


describe('Header testing', () => {
    beforeEach(() => {
        useDispatch.mockReturnValue(() => () => { });
    })
    afterEach(() => {
        jest.clearAllMocks();
    });
    test('Snapshot testing', () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    
})