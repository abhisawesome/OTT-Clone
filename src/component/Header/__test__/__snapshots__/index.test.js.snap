// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testing Snapshot testing 1`] = `
<div
  className="flex flex-row items-center"
>
  <div>
    <img
      alt="logo"
      className="h-5 w-5 sm:pl-30"
      src="images/Back.png"
    />
  </div>
  <div>
    <img
      alt="logo"
      className="h-5 w-5 pl-100"
      src="/images/search.png"
    />
  </div>
  <div>
    <input
      className="text-black"
      onChange={[Function]}
      placeholder="Search movie.."
      type="text"
    />
  </div>
</div>
`;

exports[`Header testing case 1 1`] = `
<div
  className="flex flex-row items-center"
>
  <div>
    <img
      alt="logo"
      className="h-5 w-5 sm:pl-30"
      src="images/Back.png"
    />
  </div>
  <div>
    <img
      alt="logo"
      className="h-5 w-5 pl-100"
      src="/images/search.png"
    />
  </div>
  <div>
    <input
      className="text-black"
      onChange={[Function]}
      placeholder="Search movie.."
      type="text"
    />
  </div>
</div>
`;
